packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-model.ts

```diff
@@ -0,0 +1,239 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+import { IContainerDefinition, VpcConfig } from './base-types';
+
+/**
+ * Properties for creating an Amazon SageMaker model
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-sagemaker.html
+ * @experimental
+ */
+export interface SageMakerCreateModelProps extends sfn.TaskStateBaseProps {
+  /**
+   * An execution role that you can pass in a CreateModel API request
+   *
+   * @default - a role will be created.
+   */
+  readonly role?: iam.IRole;
+  /**
+   * The name of the new model.
+   */
+  readonly modelName: string;
+  /**
+   * The definition of the primary docker image containing inference code, associated artifacts,
+   * and custom environment map that the inference code uses when the model is deployed for predictions.
+   */
+  readonly primaryContainer: IContainerDefinition;
+  /**
+   * Specifies the containers in the inference pipeline.
+   *
+   * @default - None
+   */
+  readonly containers?: IContainerDefinition[];
+
+  /**
+   * Isolates the model container. No inbound or outbound network calls can be made to or from the model container.
+   *
+   * @default false
+   */
+  readonly enableNetworkIsolation?: boolean;
+
+  /**
+   * A VpcConfig object that specifies the VPC that you want your model to connect to.
+   *
+   * @default - None
+   */
+  readonly vpcConfig?: VpcConfig;
```

Comment:
why not an `IVpc` - I don't believe we use the config type anywhere in the CDK.
Is there anything that's required that isn't available in `IVpc`?

Reply:
`IVpc` by itself doesn't provide the opportunity to select subnets.
VpcConfig defines `ec2.IVpc` and the selection of the subnets `ec2.SubnetSelection` to use. If we were to remove the option of specifying the subnets to choose, we limit the end user options
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts#L241
```
/**
 * Specifies the VPC that you want your Amazon SageMaker training job to connect to.
 *
 * @experimental
 */
export interface VpcConfig {
  /**
   * VPC
   */
  readonly vpc: ec2.IVpc;

  /**
   * VPC subnets.
   *
   * @default - Private Subnets are selected
   */
  readonly subnets?: ec2.SubnetSelection;
}
```
https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_VpcConfig.html expects SecurityGroupIds and Subnets.
`SagemakerCreateTrainingJob` https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-training-job.ts#L79 has the same input and it will make sense for `SageMakerCreateModel` to have similar input unless there is a better way to refactor them.

Reply:
I've edited the task to consume ec2.IVpc and ec2.SubnetSelection directly

Reply:
thanks for that, it seems that's closer in alignment with how we take that data in on our stable modules so I think we'll be better for having that consistency right now.